components/count-badge/count-badge-mixin.js

```diff
@@ -157,9 +157,6 @@ export const CountBadgeMixin = superclass => class extends LocalizeCoreElement(R
 	renderCount(numberStyles) {
 		let numberString = `${this.number}`;
 		const hideNumber = this.hideZero && this.number === 0;
-		if (hideNumber) {
-			numberString = '';
-		}
 		numberStyles = {
 			...numberStyles,
 			visibility: hideNumber ? 'hidden' : 'visible'
```

Comment:
The number will still be invisible, but take up the correct height for the badge, so if the count goes to zero, the layouts are not affected.